unit_test/onion_requests.cpp

```diff
@@ -0,0 +1,117 @@
+#include <boost/test/unit_test.hpp>
+#include <iostream>
+#include <ostream>
+
+#include "onion_processing.h"
+
+using namespace oxen;
+
+BOOST_AUTO_TEST_SUITE(onion_requests)
+
+constexpr const char* plaintext = "plaintext";
+constexpr const char* ciphertext = "ciphertext";
+
+// Provided "headers", so the request terminates
+// at a service node.
+BOOST_AUTO_TEST_CASE(final_destination) {
+    const std::string inner_json = R"#({
+        "headers": "something"
+    })#";
+
+    CiphertextPlusJson combined{ ciphertext, inner_json };
+
+    auto res = process_inner_request(combined, plaintext);
+
+    auto expected = FinalDestinationInfo {
+        ciphertext
+    };
+
+    BOOST_CHECK_EQUAL(std::holds_alternative<FinalDestinationInfo>(res), true);
```

Comment:
```suggestion
    BOOST_REQUIRE(std::holds_alternative<FinalDestinationInfo>(res));
```

Reply:
Good to know. I assumed `BOOST_CHECK_EQUAL` also aborted the test.

Reply:
There's a BOOST_REQUIRE_EQUAL as well, but the `== true` part seemed sort of weird.